grouper/util.py

```diff
@@ -6,19 +6,22 @@
 from typing import TYPE_CHECKING, TypeVar
 
 if TYPE_CHECKING:
+    from argparse import Namespace
     from grouper.settings import Settings
     from typing import Any, Callable, Dict, List, Optional, Pattern
 
-T_Co = TypeVar("T_Co", covariant=True)
+T = TypeVar("T")
 
 _TRUTHY = {"true", "yes", "1", ""}
 
 
 def qp_to_bool(arg):
+    # type: (str) -> bool
     return arg.lower() in _TRUTHY
 
 
 def get_loglevel(args, base=None):
+    # type: (Namespace, int) -> int
```

Comment:
I believe `base` should be `Optional[int]`

Reply:
Good catch, indeed.